sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -305,6 +305,51 @@ object GpuDeviceManager extends Logging {
     }
   }
 
+  /** Visible for GpuPythonHelper */
```

Comment:
nit: Thanks for adding this. I would suggest to improve this doc so that it explains the expected input and output of this function as well. It would be nice especially if it explains how it handles when `spark.rapids.memory.gpu.pool` and `spark.rapids.memory.gpu.pooling.enabled` conflicts.

Reply:
Personally `spark.rapids.memory.gpu.pooling.enabled` being deprecated means you do not have to set it manually and it will be true by default. Then `spark.rapids.memory.gpu.pool` will take effect to control the pool mode.

So for users, we recommond the new conf, not use them both as the same time.

Reply:
I'm not sure what you mean here, my comment was to suggest improving the scala doc for this function you added.